docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,543 @@
+  clusterIPs to be reachable across UDNs.
+    * If we don’t want to allow ALL clusterIPs to be exposed to other UDNs,
+      then on a per-service level in addition to the UDN CR requesting for it,
+      we can add a granularity level of “making that service be a candidate for UDNConnectivity”
+* If `networkPolicyEnforcement` is enabled then network policies will span across UDNs.
+   * If we want to only allow some policies then again label based granularity will be needed.
+* `connectSubnet` field is used to take a configurable subnet from the end user that will
+  be used to connect the different networks together. The slice needs to be big enough to
+  accommodate an IP per node per network. So if there are N nodes in your cluster and M
+  networks you should have N*M IPs for allocation in this slice. This value is expected
+  to be configured uniquely across multiple UDNConnect CRs that select the same network
+  i.e. If a given network is selected as part of more than 1 connect API, each of those
+  connect APIs must have a unique subnet. If a user accidentally configures overlapping
+  values, we must have a status check to block and report this in the status (details to be fleshed out).
+
+#### API Validations
+
+
+#### Valid ways of Connecting/Selecting UDNs
+
+TBD
+* A network could be selected as part of more than one network, so what is the expectation
+  when two Connect APIs are designed that select the exact same set of networks? How do
+  we control validation of such scenarios?
+* Or when a second connect API selects a subset of networks in the other connect API -> just
+  ECMP connections? Or duplication on OVN layer? Or we want to prevent this - this is a hard
+  problem -> I guess at the end of the day this problem exists for all KAPIs we just need rules
+  to be additive.
+
+### Implementation Details
+
+#### IP Address Management for Connected Networks
+
+
+#### Controller Design
+
+We will add: <--> TBD some of these bits depend on which design we pick (scroll to the Topology section) -> Idea2 will really simplify things
+* A new connect-subnet-manager (TBD maybe a controller really?) running only on ovnkube-control-plane
+  i.e cluster-manager pods that will do:
+  * The per node `connectSubnet` slice allocation according to the set
+    `connectSubnet.hostLength` field. A slice of the connect subnet will be handed
+    out to each node and annotated on the node using a new annotation
+    `k8s.ovn.org/node-connect-udn-subnets` that stores the value for each connect
+    UDN API for that node. This mechanism will be very similar to the
+    `k8s.ovn.org/node-subnets` annotation behaviour we have today.
+  * Per node-network router tunnel-key allocation that the topology design
+    (see further sections) needs since there is a requirement of no
+    conflicts of tunnel-keys across nodes. Each connect API would need N*M tunnel keys
+    if N is the number of nodes and M is the number of networks to be connected together.
```

Comment:
i dont see why you need a tunnel key. The UDNs themselves are already connected across nodes. The routing should be done within a node for connected UDNs and never across nodes imo.

Reply:
oh nvm now  i see what idea 1 and 2 are once I look at the rendered file :) Just to state my preference here I think 2 is better

Reply:
yup lets talk about it today in the meeting